### ISSUE TITLE
GUI markup script

### ISSUE THREAD

**Issue by arakov** [ELENA-LANG/elena-lang#315]
A new UI library to be similar to .NET WPF

**Comment by arakov** [comment]
It will use DirectX for Windows version and potentially OpenGL for Linux

**Comment by bencz** [comment]
The DX or OpenGL will be used to drawn buttons, menus, textbox and etc ?

**Comment by arakov** [comment]
yes, exactly like in WPF.

**Comment by bencz** [comment]
I see... This will be really interesting!! I'm ready to help with whatever is possible

**Comment by bencz** [comment]
This can be usefull: http://www.chadvernon.com/blog/resources/managed-directx-2/creating-a-gui-part-1/

**Comment by arakov** [comment]
Yes, I have seen it. Unfortunately it is in C#, I need native library. I found this - https://github.com/Microsoft/DirectX-Graphics-Samples

**Comment by bencz** [comment]
Oh.... ok, I found this: https://github.com/ocornut/imgui

**Comment by arakov** [comment]
thanks, I will look

**Comment by bencz** [comment]
Will the system support something like XAML?

**Comment by arakov** [comment]
Yes, I started this exactly for that reason: declarative UI

**Comment by bencz** [comment]
I see...
Will be interesting...

<button>
    <text>Click-me</text>
    <event>{{ call elena method }}</event>
</button>

The event part will be interesting.... will be used something like the ecode ?
Or will be possible to call a method directly ?

**Comment by arakov** [comment]
It will use ELENA Script Engine. So for the dynamic version it will create a dynamic tape, for the static one it will compile it directly
